ffi/build.rs

```diff
@@ -41,10 +59,61 @@ fn main() {
                 }
             }
         }
-    } else {
-        fs::remove_file("mitmproxy_rs/windows-redirector.exe").ok();
-        for wd_file in windivert_files {
-            fs::remove_file(format!("mitmproxy_rs/{wd_file}")).ok();
+    }
+
+    #[cfg(target_os = "macos")]
+    {
+        // macos-certificate-truster binary
+        copy_dir(
+            Path::new("../macos-certificate-truster/macos-certificate-truster.app/"),
+            Path::new("mitmproxy_rs/macos-certificate-truster.app"),
+        );
+        if fs::create_dir_all("mitmproxy_rs/macos-certificate-truster.app/Contents/MacOS/").is_ok() && cfg!(debug_assertions) {
```

Comment:
Feels like `fs::create_dir_all` should not be part of the if clause?

Reply:
`/MacOS/` is an **empty** directory because the binary is added after this line, so github doesn't add it to the repo, we have to create it to avoid `panic`, I don't know if this is the best way to create/check the directory but `fs::create_dir_all()` returns a `Result` so I thought to put it inside the if condition with `.is_ok()`.

Reply:
Yes, all correct. My point was that if `fs::create_dir_all("mitmproxy_rs/macos-certificate-truster.app/Contents/MacOS/").is_ok()` evaluates to false, we will hit the else clause (line 77+), which will not work either. 😅 

```rust
fs::create_dir_all("mitmproxy_rs/macos-certificate-truster.app/Contents/MacOS/").unwrap()
if cfg!(debug_assertions) { 
    ...
```

is perfectly fine here IMO. You can make it `.expect` instead of `unwrap` if that makes you feel better. 😉

Reply:
yes, it makes perfect sense